modules/000-common/images/vxlan-offloading-fixer/main.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"regexp"
+)
+
+var (
+	defaultPathToEthtool = "/opt/ethtool"
+)
+
+func main() {
+	pathToEthtool := defaultPathToEthtool
+	if path := os.Getenv("ETHTOOL_PATH"); path != "" {
+		pathToEthtool = path
+	}
+
+	// logger := slog.New(slog.NewJSONHandler(os.Stdout, nil))
+	// slog.SetDefault(logger)
+
+	driverRegexp := regexp.MustCompile(`^driver:\s*(\s+)\s.B`)
+	udpSegmentRegexp := regexp.MustCompile(`^tx-udp_tnl-segmentation:\s*(\s+)\s.B`)
+	udpChsumSegmentRegexp := regexp.MustCompile(`^tx-udp_tnl-csum-segmentation:\s*(\s+)\s.B`)
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		log.Fatalf("could not get network interfaces: %+v\n", err.Error())
+	}
+
+	log.Println("fix offloading for network interfaces...")
+	for _, iface := range ifaces {
+		if iface.Name == "lo" {
```

Comment:
cilium_XXX...

Reply:
this is a redundant check. only interfaces with the vmxnet3 driver are processed. cilium_* interfaces have it veth driver

Reply:
Let's refactor it. Pass the Pod IP via ENV and find the interface with this IP. It will be the local interface we must fix.